Service/OS/AndroidFCMNotification.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace RMS\PushNotificationsBundle\Service\OS;
+
+use Psr\Log\LoggerInterface;
+use RMS\PushNotificationsBundle\Exception\InvalidMessageTypeException,
+    RMS\PushNotificationsBundle\Message\AndroidMessage,
+    RMS\PushNotificationsBundle\Message\MessageInterface;
+use Buzz\Browser,
+    Buzz\Client\AbstractCurl,
+    Buzz\Client\Curl,
+    Buzz\Client\MultiCurl;
+
+class AndroidFCMNotification implements OSNotificationServiceInterface
+{
+
+    /**
+     * FCM endpoint
+     *
+     * @var string
+     */
+    protected $apiURL = "https://fcm.googleapis.com/fcm/send";
+
+    /**
+     * Google FCM API key
+     *
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * Max registration count
+     *
+     * @var integer
+     */
+    protected $registrationIdMaxCount = 1000;
+
+    /**
+     * Browser object
+     *
+     * @var \Buzz\Browser
+     */
+    protected $browser;
+
+    /**
+     * Collection of the responses from the FCM communication
+     *
+     * @var array
+     */
+    protected $responses;
+
+    /**
+     * Monolog logger
+     *
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * Constructor
+     *
+     * @param string       $apiKey
+     * @param bool         $useMultiCurl
+     * @param int          $timeout
+     * @param LoggerInterface $logger
+     * @param AbstractCurl $client (optional)
+     */
+    public function __construct($apiKey, $useMultiCurl, $timeout, $logger, AbstractCurl $client = null)
+    {
+        $this->apiKey = $apiKey;
+        if (!$client) {
+            $client = ($useMultiCurl ? new MultiCurl() : new Curl());
+        }
+        $client->setTimeout($timeout);
+
+        $this->browser = new Browser($client);
+        $this->browser->getClient()->setVerifyPeer(false);
+        $this->logger = $logger;
+    }
+
+    /**
+     * Sends the data to the given registration IDs via the FCM server
+     *
+     * @param  \RMS\PushNotificationsBundle\Message\MessageInterface              $message
+     * @throws \RMS\PushNotificationsBundle\Exception\InvalidMessageTypeException
+     * @return bool
+     */
+    public function send(MessageInterface $message)
+    {
+        if (!$message instanceof AndroidMessage) {
+            throw new InvalidMessageTypeException(sprintf("Message type '%s' not supported by FCM", get_class($message)));
+        }
+        if (!$message->isFCM()) {
+            throw new InvalidMessageTypeException("Non-FCM messages not supported by the Android FCM sender");
+        }
+
+        $headers = array(
+            "Authorization: key=" . $this->apiKey,
+            "Content-Type: application/json",
+        );
+        $data = array_merge(
+            $message->getFCMOptions(),
+            array("data" => $message->getData())
```

Comment:
I checked with the FCM documentation and also tried it out. The Key here must be called "notification" instead of the old "data"

Reply:
and also in this case of FCM the $message->getData() method should return an array with the 'body' key insead of 'message'.